src/ReturnBookControl.java

```diff
@@ -1,77 +1,72 @@
-	private loan currentLoan;
-	
+    private ReturnBookUI ui;
+    private enum CONTROL_STATE {INITIALISED, READY, INSPECTING};
+    private CONTROL_STATE state;
+    private Library library;
+    private Loan currentLoan;
 
-	public ReturnBookControl() {
-		this.library = library.INSTANCE();
-		state = CONTROL_STATE.INITIALISED;
-	}
-	
-	
-	public void setUI(ReturnBookUI ui) {
-		if (!state.equals(CONTROL_STATE.INITIALISED)) {
-			throw new RuntimeException("ReturnBookControl: cannot call setUI except in INITIALISED state");
-		}	
-		this.ui = ui;
-		ui.setState(ReturnBookUI.UI_STATE.READY);
-		state = CONTROL_STATE.READY;		
-	}
+    
+    public ReturnBookControl() {
+        this.library = Library.INSTANCE();
+        state = CONTROL_STATE.INITIALISED;
+    }
 
+    
+    public void setUI(ReturnBookUI ui) {
+        if (!state.equals(CONTROL_STATE.INITIALISED)) {
+            throw new RuntimeException("ReturnBookControl: cannot call setUI except in INITIALISED state");
+        }
+        this.ui = ui;
+        ui.setState(ReturnBookUI.UI_STATE.READY);
+        state = CONTROL_STATE.READY;
+    }
 
-	public void bookScanned(int bookId) {
-		if (!state.equals(CONTROL_STATE.READY)) {
-			throw new RuntimeException("ReturnBookControl: cannot call bookScanned except in READY state");
-		}	
-		book currentBook = library.Book(bookId);
-		
-		if (currentBook == null) {
-			ui.display("Invalid Book Id");
-			return;
-		}
-		if (!currentBook.On_loan()) {
-			ui.display("Book has not been borrowed");
-			return;
-		}		
-		currentLoan = library.getLoanByBookId(bookId);	
-		double overDueFine = 0.0;
-		if (currentLoan.isOverDue()) {
-			overDueFine = library.calculateOverDueFine(currentLoan);
-		}
-		ui.display("Inspecting");
-		ui.display(currentBook.toString());
-		ui.display(currentLoan.toString());
-		
-		if (currentLoan.isOverDue()) {
-			ui.display(String.format("\nOverdue fine : $%.2f", overDueFine));
-		}
-		ui.setState(ReturnBookUI.UI_STATE.INSPECTING);
-		state = CONTROL_STATE.INSPECTING;		
-	}
+    
+    public void bookScanned(int bookId) {
+        if (!state.equals(CONTROL_STATE.READY)) {
+            throw new RuntimeException("ReturnBookControl: cannot call bookScanned except in READY state");
+        }
+        Book currentBook = library.book(bookId);
+        if (currentBook == null) {
+            ui.display("Invalid Book Id");
+            return;
+        }
+        if (!currentBook.On_loan()) {
+            ui.display("Book has not been borrowed");
+            return;
+        }
+        currentLoan = library.getLoanByBookId(bookId);
+        double overDueFine = 0.0;
+        if (currentLoan.isOverDue()) {
+            overDueFine = library.calculateOverDueFine(currentLoan);
+        }
+        ui.display("Inspecting");
+        ui.display(currentBook.toString());
+        ui.display(currentLoan.toString());
+        if (currentLoan.isOverDue()) {
+            ui.display(String.format("\nOverdue fine : $%.2f", overDueFine));
```

Comment:
D4 - Change this line to remove method call within argument list of method call e.g.             
48 String overDueFineMessage = String.format("\nOverdue fine : $%.2f", overDueFine);
49 ui.display(overDueFineMessage);